pkg/controller/vsphere/util.go

```diff
@@ -0,0 +1,82 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+
+	"gopkg.in/gcfg.v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	globalInfrastuctureName  = "cluster"
+	openshiftConfigNamespace = "openshift-config"
+)
+
+// vSphereConfig is a copy of the Kubernetes vSphere cloud provider config type
+// that contains the fields we need.  Unfortunately, we can't easily import
+// either the legacy or newer cloud provider code here, so we're just
+// duplicating part of the type and parsing it ourselves using the same gcfg
+// library for now.
+type vSphereConfig struct {
+	Labels struct {
+		Zone   string `gcfg:"zone"`
+		Region string `gcfg:"region"`
+	}
+}
+
+func getInfrastructure(client runtimeclient.Client) (*configv1.Infrastructure, error) {
+	infra := &configv1.Infrastructure{}
+	infraName := runtimeclient.ObjectKey{
+		Name: globalInfrastuctureName,
+	}
+
+	if err := client.Get(context.Background(), infraName, infra); err != nil {
+		return nil, err
+	}
+
+	return infra, nil
+}
+
+func getVSphereConfig(client runtimeclient.Client) (*vSphereConfig, error) {
+	infra, err := getInfrastructure(client)
+	if err != nil {
```

Comment:
do you think we could skip this and assume the given name for the configmap -> https://github.com/openshift/installer/blob/a76a6b34092d99835d0e947ae9343d231f38f7e1/pkg/asset/manifests/cloudproviderconfig.go#L80

Reply:
We could, but I think this is not too complicated, and it would be less correct to hard code the name.